File: Data Structures & Algorithms/validate-parentheses/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        if(s.length%2 !== 0) return false;
        let last;
        let stack = [];
        const map = new Map([
            ['(',')'],
            ['{','}'],
            ['[',']'],
        ])
        for(let i = 0; i < s.length ; i++){
            if(s[i] == '(' || s[i] == '{' || s[i] == '[')
                stack.push(s[i])
            else{
                last = stack.pop();
                console.log("Last: ", last)
                console.log("MAP LAST ", map.get(last))
                console.log("s[i] ", s[i])
                if(s[i] !== map.get(last))
                    return false
            }
                
        }

        return stack.length === 0;
    }
}

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    // public int maxDepth(TreeNode root) {
    //     if (root != null) return 0;

    //     int left = 0; int right = 0; int max = 0;
    //     Deque<TreeNode> stack = new ArrayDeque<>();
    //     TreeNode curr = root;

    //     while(curr != null || !stack.isEmpty()){
    //         while(curr != null){
    //             stack.push(curr);
    //             curr = curr.left;
    //         }

    //         curr = stack.pop();
    //         max += 1;
    //     }
    // }


    public int maxDepth(TreeNode root) {
        if (root == null) return 0;
        int left = 0; int right = 0; 
        
        left = 1 + maxDepth(root.left);
        right = 1 + maxDepth(root.right);

        return Math.max(left, right);
    }
}
